Flow.Launcher.Plugin/Interfaces/IPublicAPI.cs

```diff
@@ -298,5 +299,81 @@ public interface IPublicAPI
+        /// that returns a result.
+        /// </summary>
+        /// <param name="messageBoxText">A System.String that specifies the text to display.</param>
+        /// <param name="caption">A System.String that specifies the title bar caption to display.</param>
+        /// <param name="button">
+        /// A System.Windows.MessageBoxButton value that specifies which button or buttons
+        /// to display.
+        /// </param>
+        /// <returns>
+        /// A System.Windows.MessageBoxResult value that specifies which message box button
+        /// is clicked by the user.
+        /// </returns>
+        public MessageBoxResult ShowMsgBox(string messageBoxText, string caption, MessageBoxButton button); 
+
+        /// <summary>
+        /// Displays a message box that has a message, title bar caption, button, and icon;
+        /// and that returns a result.
+        /// </summary>
+        /// <param name="messageBoxText">A System.String that specifies the text to display.</param>
+        /// <param name="caption">A System.String that specifies the title bar caption to display.</param>
+        /// <param name="button">
+        /// A System.Windows.MessageBoxButton value that specifies which button or buttons
+        /// to display.
+        /// </param>
+        /// <param name="icon">A System.Windows.MessageBoxImage value that specifies the icon to display.</param>
+        /// <returns>
+        /// A System.Windows.MessageBoxResult value that specifies which message box button
+        /// is clicked by the user.
+        /// </returns>
+        public MessageBoxResult ShowMsgBox(string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon);
+
+        /// <summary>
+        /// Displays a message box that has a message, title bar caption, button, and icon;
+        /// and that accepts a default message box result and returns a result.
+        /// </summary>
+        /// <param name="messageBoxText">A System.String that specifies the text to display.</param>
+        /// <param name="caption">A System.String that specifies the title bar caption to display.</param>
+        /// <param name="button">
+        /// A System.Windows.MessageBoxButton value that specifies which button or buttons
+        /// to display.
+        /// </param>
+        /// <param name="icon">A System.Windows.MessageBoxImage value that specifies the icon to display.</param>
+        /// <param name="defaultResult">
+        /// A System.Windows.MessageBoxResult value that specifies the default result of
+        /// the message box.
+        /// </param>
+        /// <returns>
+        /// A System.Windows.MessageBoxResult value that specifies which message box button
+        /// is clicked by the user.
+        /// </returns>
+        public MessageBoxResult ShowMsgBox(string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult);
```

Comment:
Maybe instead of four methods, can just do one with defaults?

Reply:
I just follow the same style as the System.Windows.MessageBox. Do you think it is necessary?

Reply:
@jjw24 Any ideas?

Reply:
Yeah following that style is good. I just meant we can just have one method and have caption (empty string), button (null) and etc set with defauls. Let PublicAPIInstance.cs handle how to call based on what parameters are passed in.

Reply:
Get it.